src/PhpSpreadsheet/Calculation/Engineering.php

```diff
@@ -2423,6 +2423,175 @@ public static function erfVal($x)
         return self::$twoSqrtPi * $sum;
     }
 
+    protected static function validateBitwiseArgument($value)
```

Comment:
I think this should be private to limit our API surface and make future maintenance easier. Also maybe a doc block would be nice to document types.

Also could/should the Functions::flattenSingleValue() call be part of this method ?

Reply:
Valid comments; and I've also tightened up the validation to handle the different responses for different errors. See my next commit.
I'm very much tempted to write a dedicated validator class, that can be called passing in the argument with a set of validation rules/responses; but that's a more significant piece of work that really needs its own PR rather than bundling with this one.